### ISSUE TITLE
classpath: resources do not work with SpringLiquibase when test rollback on update is enabled

### ISSUE THREAD

**Issue by wilkinsona** [liquibase/liquibase#4135]
### Search first

- [X] I searched and no similar issues were found

### Description

There appears to be a regression in Liqubase 4.21.0 that breaks database migration when test rollback on update is enabled. The failure is the following:

```
Exception in thread "main" org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'springLiquibase' defined in com.example.springliquibaseregression.SpringLiquibaseRegression: Invocation of init method failed; nested exception is liquibase.exception.CommandExecutionException: liquibase.exception.ChangeLogParseException: classpath:db/changelog.yaml does not exist
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1804)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:620)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:955)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:920)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583)
	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:93)
	at com.example.springliquibaseregression.SpringLiquibaseRegression.main(SpringLiquibaseRegression.java:23)
Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.ChangeLogParseException: classpath:db/changelog.yaml does not exist
	at liquibase.command.CommandScope.execute(CommandScope.java:235)
	at liquibase.Liquibase.rollback(Liquibase.java:644)
	at liquibase.Liquibase.updateTestingRollback(Liquibase.java:1212)
	at liquibase.Liquibase.updateTestingRollback(Liquibase.java:1197)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:323)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:284)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1863)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1800)
	... 11 more
Caused by: liquibase.exception.ChangeLogParseException: classpath:db/changelog.yaml does not exist
	at liquibase.parser.core.yaml.YamlChangeLogParser.parse(YamlChangeLogParser.java:31)
	at liquibase.command.core.helpers.DatabaseChangelogCommandStep.getDatabaseChangeLog(DatabaseChangelogCommandStep.java:100)
	at liquibase.command.core.helpers.DatabaseChangelogCommandStep.run(DatabaseChangelogCommandStep.java:77)
	at liquibase.command.CommandScope.execute(CommandScope.java:212)
	... 18 more
```

Debugging has shown that `YamlChangeLogParser.parse` is called twice – once with `SpringResourceAccessor` which works and once with `ClassLoaderResourceAccessor` which fails. Upon downgrading to 4.20.0, `parse` is only called once with `SpringResourceAccessor` so the failure does not occur.

### Steps To Reproduce

- Download [spring-liquibase-regression.zip](https://github.com/liquibase/liquibase/files/11231972/spring-liquibase-regression.zip) and unzip it
- Import the resulting Gradle project into your IDE
- Run the main method of `SpringLiquibaseRegression`

### Expected/Desired Behavior

The behavior is as it was in 4.20.x and no failure occurs when using SpringLiquibase with test rollback on update enabled.

### Liquibase Version

4.21.0

### Database Vendor & Version

Any

### Liquibase Integration

Spring

### Liquibase Extensions

None

### OS and/or Infrastructure Type/Provider

Any

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by snicoll** [comment]
FTR This prevents us to upgrade to a more recent Liquibase version in Spring Boot.

**Comment by EvaristeGalois11** [comment]
I just tried to reproduce your problem and with liquibase 4.23.0 it seems to be already fixed.

I did a git bisect and this was the PR that indirectly fixed it #4282.

Could you @wilkinsona or @snicoll verify it too?

**Comment by wilkinsona** [comment]
Thanks very much, @EvaristeGalois11. I think you're right. With 4.23.0, I can't reproduce the failure we were seeing with the earlier versions. I'll close this as a duplicate of #4282.
